AEPCampaignClassic/Sources/CampaignClassic.swift

```diff
@@ -46,27 +47,112 @@ public class CampaignClassic: NSObject, Extension {
+            return false
+        }
+        return configurationSharedState.status == .set
     }
 
     /// Handles `Configuration Response` events
     /// - Parameter event: the Configuration `Event` to be handled
     private func handleConfigurationEvents(event: Event) {
+        Log.trace(label: CampaignClassicConstants.LOG_TAG, "An event of type '\(event.type)' has been received.")
+        dispatchQueue.async { [weak self] in
+            // todo handle config update
+        }
+    }
+
+    /// Handles CampaignClassic's registerDevice, track notification click and track notification receive events.
+    /// All the other the campaign events are ignored
+    /// - Parameter event: the  Campaign Request content event to be handled
+    private func handleCampaignEvents(event: Event) {
         Log.trace(label: CampaignClassicConstants.LOG_TAG, "An event of type '\(event.type)' has been received.")
         dispatchQueue.async { [weak self] in
             guard let self = self else {return}
-            self.updateCampaignState(event: event)
-            // update CampaignClassicState
-            if self.state.privacyStatus == PrivacyStatus.optedOut {
-                // handle opt-out
-                return
+            if event.isRegisterEvent {
+                self.handleRegisterDeviceEvent(event: event)
+            } else if event.isTrackClickEvent {
+                self.handleTrackEvent(event: event, withTagId: CampaignClassicConstants.TRACK_CLICK_TAGID)
+            } else if event.isTrackReceiveEvent {
+                self.handleTrackEvent(event: event, withTagId: CampaignClassicConstants.TRACK_RECEIVE_TAGID)
             }
         }
     }
 
-    /// Updates the `CampaignClassicState` with the shared state of other required extensions
-    /// - Parameter event: the `Event`containing the shared state of other required extensions
-    private func updateCampaignState(event: Event) {
-        // todo
+    private func handleRegisterDeviceEvent(event: Event) {
+        // todo - coming soon
+    }
+
+    /// Sends a track request to the configured Campaign Classic tracking server upon notification receive or click.
+    /// If Configuration is not available or Campaign Classic is not configured, no track request shall be sent.
+    /// Tracking identifiers messageId(_mId) and deliveryId (_dId), retrieved from the message payload, are
+    /// required for track request to be sent.
+    ///
+    /// - Parameters:
+    ///   - event : the incoming track event
+    ///   - tagId : an integer string indicating whether it is a notification receive or notification click request
+    private func handleTrackEvent(event: Event, withTagId tagId: String) {
+        let configuration = CampaignClassicConfiguration.init(forEvent: event, runtime: runtime)
+
+        guard let trackingServer = configuration.trackingServer, !trackingServer.isEmpty else {
```

Comment:
we could probably move the logic of checking for empty string to the getters in your `CampaignClassicConfiguration` class - would make this code a lot cleaner